castle/cms/tiles/calendar.py

```diff
@@ -23,10 +24,60 @@ def format_date(dt):
             pass
 
 
+class ICalendarTileSchema(Interface):
+
+    form.widget(query=QueryFieldWidget)
+    query = schema.List(
+        title=u'Base query',
+        description=u"This query can be customized base on user selection",
+        value_type=schema.Dict(value_type=schema.Field(),
+                               key_type=schema.TextLine()),
+        required=False,
+        default=[{
+            u'i': u'portal_type',
+            u'v': [u'Event'],
+            u'o': u'plone.app.querystring.operation.selection.any'}]
+    )
+
+    sort_on = schema.TextLine(
+        title=u'Sort on',
+        description=u"Sort on this index",
+        required=False,
+    )
+
+    sort_reversed = schema.Bool(
+        title=u'Reversed order',
+        description=u'Sort the results in reversed order',
+        required=False,
+        default=False,
+    )
+
+    limit = schema.Int(
+        title=u'Limit',
+        description=u'Limit Search Results',
+        required=False,
+        default=250,
+        min=1,
+    )
+
+
 class CalendarTile(BaseTile):
+    implements(ICalendarTileSchema)
 
     def get_query(self):
         parsed = parse_query_from_data(self.data, self.context)
+        # If parsed is empty and the self.data doesn't contain any of the variables below
+        # then we have a new calendar to set the defaults on.
+        if parsed == {}:
+            if self.data.get('query', None) is None:
+                if self.data.get('limit', None) is None:
+                    if self.data.get('sort_reversed', None) is None:
+                        if self.data.get('sort_on', None) is None:
+                            self.data[u'limit'] = ICalendarTileSchema["limit"].default
```

Comment:
Here, the idea of the defaults in the schema is that they are set on the instance (the new calendar tile in this case) by default when it is created.  Such that if the user attempted to create it with a None value, the default would be used.  So I wonder if you were seeing some different behavior than that, that would cause you to do this whole block?  With the defaults that are now in the schema, this line would not likely be reached since if no value is inputted for limit for example, limit would still be 250 on line 73 rather than None, because 250 is the default set in the schema.

Reply:
Actually for brand new calendars the parser gets nothing and as a result will give out nothing.  The if then statements are there to check if there are a bunch of none's in the self.data.  And if there is a bunch of nones in the self.data then we know that this is a brand new calendar.  Therefore initialize the self.data with defaults.